spec/lib/root_route_constraint_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+
+describe RootRouteConstraint do
+  describe '#matches?' do
+
```

Comment:
`let` initialize the `constraint` variable before the Factories are created, and hence the `constraint` variable will not be referring to the correct database configuration. `constraint = RootRouteConstraint.new` only needs to be run after the factories are created for a particular example and the creation for every example is different.